### DISCUSSION TITLE
Error authentication git push --set-upstream origin main

### DISCUSSION THREAD

**Discussion by Kumaro001** [community/community#138140] [Programming Help]
### Body

Hello all,

I'm blocked create : 
git push --set-upstream origin main  
Username for 'https://github.com': Kumaro001
Password for 'https://Kumaro001@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal : Échec d'authentification pour 'https://github.com/Kumaro001/git-kumaro001.git/'

my password is correct but i get the same message.

Could someone help me please?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment]
Hi @Kumaro001 ,

Have you generated a `Personal Access Token` to use as a password for git operations with `HTTPS`? Using the GitHub account password is no longer supported.

For more information on how to generate a `PAT` see the [docs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic):

**Comment by Thxmxs** [comment]
The error you're encountering is due to GitHub removing password-based authentication for Git operations over HTTPS. Since August 13, 2021, GitHub requires you to use a Personal Access Token (PAT) instead of a password when pushing or pulling from a repository.

Here’s how you can fix this:

Generate a Personal Access Token (PAT):

Go to your GitHub account and navigate to Settings.
Under Developer Settings, click on Personal Access Tokens.
Click on Generate new token.
Select the appropriate scopes (at least repo for repository access).
Copy the token (you will not be able to view it again).
Use the PAT instead of your password:

When prompted for a password, use the Personal Access Token instead of your GitHub password.
Update your Git configuration:

If you want to avoid entering the token each time, you can cache your credentials locally with the following command:
git config --global credential.helper cache
Or store them permanently:
git config --global credential.helper store

Push your changes again:
Try running the same command:
git push --set-upstream origin main
This should solve the authentication issue and allow you to push to your repository.
